sql/pgtap--1.1.0--1.2.0.sql

```diff
@@ -36,3 +36,50 @@ BEGIN
     RETURN thing;
 END;
 $$ LANGUAGE plpgsql;
+
+-- isnt_member_of( role, members[], description )
+CREATE OR REPLACE FUNCTION isnt_member_of( NAME, NAME[], TEXT )
+RETURNS TEXT AS $$
+DECLARE
+    extra text[];
+BEGIN
+    IF NOT _has_role($1) THEN
+        RETURN fail( $3 ) || E'\n' || diag (
+            '    Role ' || quote_ident($1) || ' does not exist'
+        );
+    END IF;
+
+    SELECT ARRAY(
+        SELECT quote_ident($2[i])
+          FROM generate_series(1, array_upper($2, 1)) s(i)
+          LEFT JOIN pg_catalog.pg_roles r ON rolname = $2[i]
+         WHERE r.oid = ANY ( _grolist($1) )
+         ORDER BY s.i
+    ) INTO extra;
+    IF extra[1] IS NULL THEN
+        RETURN ok( true, $3 );
+    END IF;
+    RETURN ok( false, $3 ) || E'\n' || diag(
+        '    Members, who should not be in ' || quote_ident($1) || E' role:\n        ' ||
+        array_to_string( extra, E'\n        ')
+    );
```

Comment:
Can probably be shortened to

```suggestion
    RETURN ok( extra[1] IS NULL, $3 ) || E'\n' || diag(
        '    Members, who should not be in ' || quote_ident($1) || E' role:\n        ' ||
        array_to_string( extra, E'\n        ')
    );
```

Reply:
Just tried that - failed the tests. Seems like `is_member_of()` did it for a reason.

This will - independently of having an error or not - output `     Members, who should not be in meanies role:`. And that makes sense - because, the condition passed as the first argument to `ok()` would only apply for the 2nd argument to `ok()` - not for the remaining concat in this `RETURN`. Putting it inside the `ok()` defeats the purpose of `diag()` in the first place, I guess.

Reply:
> Just tried that - failed the tests. Seems like `is_member_of()` did it for a reason.

Oh, right, because I was mis-reading `||` as "or" rather than "concat".